mock/src/components/Commands.ts

```diff
@@ -0,0 +1,106 @@
+import { REPLFunction } from "./REPLFunction";
+import { Dispatch, SetStateAction, useState } from "react";
+import { mockedDataMap, mockedResultsMap } from "./mockedJson";
+import { HistoryEntry } from "./REPL";
+
+// An export that REPLInput can use to map certain commands to functions
+export const commandMap = new Map<string, REPLFunction>();
+
+// Create a varaible to store the file
+let loadedData: string[][];
+let file: { data: string[][]; hasHeaders: boolean } | undefined;
+
+export function addCommand(
+  commandName: string,
+  commandFunction: REPLFunction
+): void {
+  commandMap.set(commandName, commandFunction);
+}
+
+export function removeCommand(commandName: string): void {
+  commandMap.delete(commandName);
+}
+
+// Initialize the basic commands
+export function setCommandMap() {
+  // Set initial values for commandMap
+  commandMap.set("mode", mode);
+  commandMap.set("load_file", load);
+  commandMap.set("view", view);
+  commandMap.set("search", search);
+}
+
+// The function for the "mode" command
+const mode: REPLFunction = (
+  args: Array<string>,
+  setVerbose: Dispatch<SetStateAction<boolean>>
```

Comment:
It's interesting you deal with verbose output here instead of outside the function. We did the latter. We thought it would lead to fewer chances of bugs, but I can see your method to be more organized